workflow/controller/controller.go

```diff
@@ -268,6 +268,7 @@ func (wfc *WorkflowController) periodicWorkflowGarbageCollector(stopCh <-chan st
 				oldRecords, err := wfc.offloadNodeStatusRepo.ListOldOffloads(wfc.GetManagedNamespace())
 				if err != nil {
 					log.WithField("err", err).Error("Failed to list old offloaded nodes")
+					return
```

Comment:
This will exit the loop. if there is any intermediate network issue or DB issue, `return` will exit the loop with an error message and stop the node status GC.
 Do you think the following behavior will fit for this scenario? 
 1. Implement ExponentialBackoff for the Query inside the loop instead of return
2. Restart the controller so it will check the DB connectivity while start.

Reply:
This fix is incorrect - should be continue - good spot. I'm not sure about back off etc. Adds a massive level of testing complexity for little benefit.

Reply:
how will the application behave if `wfc.offloadNodeStatusRepo.ListOldOffloads`  return the err?

Reply:
if will exit, I think this is incorrect, should continue (same applies for other lines) - I will fix